### PR TITLE
adapt core code to bam

### PR DESCRIPTION
**Pull request by adr-mo**
Changes made for https://github.com/centreon/centreon-modules/pull/692

Moving repositories implemented for BAM from "ServiceProvider dependency injector" to Symfony services

Tests are not OK, but it will be handled later on

### CODE REVIEW

**Review by abuathier — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by a-launois — CHANGES_REQUESTED** [review]

**Review by a-launois — APPROVED** [review]

**Comment by adr-mo** [comment]
Changing base branch to develop as the code changed is code adaptation only for BAM module. It should not impact the web module

File: centreon/src/Centreon/Domain/Repository/CommandRepository.php [hunk 1]
```diff
@@ -182,13 +191,13 @@ public function exportList(array $list): array
...
-TRUNCATE TABLE `command_categories`;
-TRUNCATE TABLE `on_demand_macro_command`;
-SQL;
+            TRUNCATE TABLE `command`;
+            TRUNCATE TABLE `connector`;
+            TRUNCATE TABLE `command_arg_description`;
+            TRUNCATE TABLE `command_categories_relation`;
+            TRUNCATE TABLE `command_categories`;
+            TRUNCATE TABLE `on_demand_macro_command`;
+        SQL;
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/CommandRepository.php:200]:
```suggestion
            SQL;
```

**Review by abuathier — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

File: centreon/src/Centreon/Domain/Repository/AclGroupRepository.php [hunk 2]
```diff
@@ -21,17 +21,41 @@
...
 
+    /**
+     * @var int $resultCountForPagination
+     */
+    private int $resultCountForPagination = 0;
+
+    /**
+     * @var array $concordanceArray
+     */
+    private array $concordanceArray = [
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/AclGroupRepository.php:43]: Immutability ? `const` / `readonly` ?

File: centreon/src/Centreon/Domain/Repository/AclGroupRepository.php [hunk 3]
```diff
@@ -47,73 +71,99 @@ public static function entityClass(): string
...
 
     /**
      * {@inheritdoc}
      */
     public function getPaginationList($filters = null, int $limit = null, int $offset = null, $ordering = []): array
     {
-        $sql = 'SELECT SQL_CALC_FOUND_ROWS t.* FROM `' . $this->getClassMetadata()->getTableName() . '` AS `t`';
+        $request = <<<SQL
+            SELECT SQL_CALC_FOUND_ROWS t.* FROM `:db`.`acl_groups` AS `t`
+        SQL;
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/AclGroupRepository.php:84]:
```suggestion
            SQL;
```

File: centreon/src/Centreon/Domain/Repository/AclGroupRepository.php [hunk 4]
```diff
@@ -47,73 +71,99 @@ public static function entityClass(): string
...
         $collector = new StatementCollector();
 
         $isWhere = false;
         if ($filters !== null) {
-            if (array_key_exists('search', $filters) && $filters['search']) {
-                $sql .= ' WHERE t.' . $this->getClassMetadata()->getColumn('name') . ' LIKE :search';
+            if (
+                array_key_exists('search', $filters)
+                && $filters['search']
+            ) {
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/AclGroupRepository.php:93]:
```suggestion
            if ($filters['search'] ?? false) {
```

File: centreon/src/Centreon/Domain/Repository/ContactGroupRepository.php [hunk 5]
```diff
@@ -49,46 +62,59 @@ public function getPaginationList($filters = null, int $limit = null, int $offse
...
-        $sql = 'SELECT SQL_CALC_FOUND_ROWS * FROM `' . ContactGroup::TABLE . '`';
+        $sql = 'SELECT SQL_CALC_FOUND_ROWS * FROM `:db`.contactgroup';
 
         $isWhere = false;
         if ($filters !== null) {
-            if (array_key_exists('search', $filters) && $filters['search']) {
+            if (
+                array_key_exists('search', $filters)
+                && $filters['search']
+            ) {
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/ContactGroupRepository.php:72]:
```suggestion
            if ($filters['search'] ?? false) {
```

File: centreon/src/Centreon/Domain/Repository/ContactGroupRepository.php [hunk 6]
```diff
@@ -49,46 +62,59 @@ public function getPaginationList($filters = null, int $limit = null, int $offse
...
+                    $collector->addValue($key, $id, \PDO::PARAM_INT);
                     unset($x, $id);
                 }
                 $sql .= $isWhere ? ' AND' : ' WHERE';
                 $sql .= ' `cg_id` IN (' . implode(',', $idsListKey) . ')';
             }
         }
 
-        if (!empty($ordering['field'])) {
+        if (! empty($ordering['field'])) {
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/ContactGroupRepository.php:94]: For homogeneity with previous
```suggestion
        if ($ordering['field'] ?? false) {
```

File: centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php [hunk 7]
```diff
@@ -21,15 +21,60 @@
...
 {
-     use CheckListOfIdsTrait;
+    use CheckListOfIdsTrait;
+
+    /**
+     * @var int $resultCountForPagination
+     */
+    private int $resultCountForPagination = 0;
+
+    private array $concordanceArray = [
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php:40]: Immutability ? `const` / `readonly` ?

File: centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php [hunk 8]
```diff
@@ -48,72 +93,113 @@ public function getPaginationList($filters = null, int $limit = null, int $offse
...
-            . 'FROM `' . $this->getClassMetadata()->getTableName() . '`';
+        $sql = 'SELECT SQL_CALC_FOUND_ROWS * FROM `:db`.`nagios_server`';
 
         if ($filters !== null) {
             $isWhere = false;
 
-            if (!empty($filters['search'])) {
-                $sql .= ' WHERE `' . $this->getClassMetadata()->getColumn('name') . '` LIKE :search';
+            if (! empty($filters['search'])) {
+                $sql .= ' WHERE `name` LIKE :search';
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php:102]: Consistency with other modifications of this type.
```suggestion
            if ($filters['search'] ?? false) {
                $sql .= ' WHERE `name` LIKE :search';
```

File: centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php [hunk 9]
```diff
@@ -48,72 +93,113 @@ public function getPaginationList($filters = null, int $limit = null, int $offse
...
                 $collector->addValue(':search', "%{$filters['search']}%");
                 $isWhere = true;
             }
 
-            if (array_key_exists('ids', $filters) && is_array($filters['ids'])) {
+            if (
+                array_key_exists('ids', $filters)
+                && is_array($filters['ids'])
+            ) {
```
Comment (abuathier) [inline centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php:110]: Missing check about empty array like in another location of this PR ?
```suggestion
            if (
                array_key_exists('ids', $filters)
                && is_array($filters['ids'])
                && [] !== $filters['ids']
            ) {
```

File: centreon/src/CentreonCommand/Domain/Repository/CommandRepository.php [hunk 10]
```diff
@@ -21,14 +21,26 @@
...
+use Centreon\Infrastructure\DatabaseConnection;
+use CentreonCommand\Domain\Entity\Command;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
 
-class CommandRepository extends ServiceEntityRepository implements PaginationRepositoryInterface
+class CommandRepository extends AbstractRepositoryRDB implements PaginationRepositoryInterface
 {
+    /**
+     * @var int $resultCountForPagination
+     */
```
Comment (abuathier) [inline centreon/src/CentreonCommand/Domain/Repository/CommandRepository.php:34]:
```suggestion
    /** @var int $resultCountForPagination */
```

File: centreon/src/CentreonCommand/Domain/Repository/CommandRepository.php [hunk 11]
```diff
@@ -21,14 +21,26 @@
...
+    private int $resultCountForPagination = 0;
+
+    /**
+     * @param DatabaseConnection $db
+     */
```
Comment (abuathier) [inline centreon/src/CentreonCommand/Domain/Repository/CommandRepository.php:39]:
```suggestion
    /** @param DatabaseConnection $db */
```

File: centreon/src/CentreonNotification/Domain/Repository/DependencyRepository.php [hunk 12]
```diff
@@ -35,14 +43,15 @@ class DependencyRepository extends ServiceEntityRepository
      */
     public function removeById(int $id): void
     {
-        $sql = "DELETE FROM `dependency`"
-            . " WHERE dep_id = :id";
+        $request = <<<SQL
+            DELETE FROM `:db`.`dependency` WHERE dep_id = :dependencyId
+        SQL;
```
Comment (abuathier) [inline centreon/src/CentreonNotification/Domain/Repository/DependencyRepository.php:48]:
```suggestion
            SQL;
```

File: centreon/src/CentreonNotification/Domain/Repository/EscalationRepository.php [hunk 13]
```diff
@@ -21,17 +21,30 @@
...
+use Centreon\Infrastructure\CentreonLegacyDB\Interfaces\PaginationRepositoryInterface;
 
-class EscalationRepository extends ServiceEntityRepository implements PaginationRepositoryInterface
+class EscalationRepository extends AbstractRepositoryRDB implements PaginationRepositoryInterface
 {
     use CheckListOfIdsTrait;
 
+    /**
+     * @var int $resultCountForPagination
+     */
```
Comment (abuathier) [inline centreon/src/CentreonNotification/Domain/Repository/EscalationRepository.php:37]:
```suggestion
    /** @var int $resultCountForPagination */
```

File: centreon/src/CentreonNotification/Domain/Repository/EscalationRepository.php [hunk 14]
```diff
@@ -48,33 +61,43 @@ public static function entityClass(): string
...
 
         $collector = new StatementCollector();
 
         $isWhere = false;
         if ($filters !== null) {
-            if (array_key_exists('search', $filters) && $filters['search']) {
+            if (
+                array_key_exists('search', $filters)
+                && $filters['search']
+            ) {
```
Comment (abuathier) [inline centreon/src/CentreonNotification/Domain/Repository/EscalationRepository.php:85]: Consistency with other modifications of this type.

```suggestion
            if ($filters['search'] ?? false) {
```

File: centreon/src/CentreonUser/Domain/Repository/TimeperiodRepository.php [hunk 15]
```diff
@@ -21,17 +21,30 @@
...
+use Centreon\Infrastructure\CentreonLegacyDB\Interfaces\PaginationRepositoryInterface;
 
-class TimeperiodRepository extends ServiceEntityRepository implements PaginationRepositoryInterface
+class TimeperiodRepository extends AbstractRepositoryRDB implements PaginationRepositoryInterface
 {
     use CheckListOfIdsTrait;
 
+    /**
+     * @var int $resultCountForPagination
+     */
```
Comment (abuathier) [inline centreon/src/CentreonUser/Domain/Repository/TimeperiodRepository.php:37]:
```suggestion
    /** @var int $resultCountForPagination */
```

File: centreon/src/CentreonUser/Domain/Repository/TimeperiodRepository.php [hunk 16]
```diff
@@ -57,24 +88,30 @@ public function checkListOfIds(array $ids): bool
...
                 $sql .= ' WHERE (`tp_name` LIKE :search OR `tp_alias` LIKE :search)';
                 $collector->addValue(':search', "%{$filters['search']}%");
                 $isWhere = true;
             }
 
-            if (array_key_exists('ids', $filters) && is_array($filters['ids'])) {
+            if (
+                array_key_exists('ids', $filters)
+                && is_array($filters['ids'])
+            ) {
```
Comment (abuathier) [inline centreon/src/CentreonUser/Domain/Repository/TimeperiodRepository.php:109]: Missing test about empty array like other location of this pr ?
```suggestion
            if (
                array_key_exists('ids', $filters)
                && is_array($filters['ids'])
                && [] !== $filters['ids']
            ) {
```

File: centreon/src/Centreon/Domain/Repository/AclGroupRepository.php (hunk 2 above)
Comment (adr-mo) [reply centreon/src/Centreon/Domain/Repository/AclGroupRepository.php:43]: done ✅

File: centreon/src/Centreon/Domain/Repository/AclGroupRepository.php (hunk 3 above)
Comment (adr-mo) [reply centreon/src/Centreon/Domain/Repository/AclGroupRepository.php:84]: done ✅

File: centreon/src/Centreon/Domain/Repository/AclGroupRepository.php (hunk 4 above)
Comment (adr-mo) [reply centreon/src/Centreon/Domain/Repository/AclGroupRepository.php:93]: done ✅

File: centreon/src/Centreon/Domain/Repository/ContactGroupRepository.php (hunk 6 above)
Comment (adr-mo) [reply centreon/src/Centreon/Domain/Repository/ContactGroupRepository.php:94]: done ✅

File: centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php (hunk 7 above)
Comment (adr-mo) [reply centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php:40]: done ✅

File: centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php (hunk 9 above)
Comment (adr-mo) [reply centreon/src/Centreon/Domain/Repository/NagiosServerRepository.php:110]: done ✅

File: centreon/src/CentreonNotification/Domain/Repository/DependencyRepository.php (hunk 12 above)
Comment (adr-mo) [reply centreon/src/CentreonNotification/Domain/Repository/DependencyRepository.php:48]: done ✅

**Comment by adr-mo** [comment]
@abuathier all your feedbacks from the code review were applied 👍 thanks for the review 💯

**Review by abuathier — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]
